### PR TITLE
add guarantee about remove_dir and remove_file error kinds

### PR DESCRIPTION
**Pull request by lolbinarycat**
approved in ACP https://github.com/rust-lang/libs-team/issues/433

### CODE REVIEW

**Comment by lolbinarycat** [comment]
r? libs

File: library/std/src/fs.rs [hunk 1]
```diff
@@ -1989,6 +1989,11 @@ impl AsInner<fs_imp::DirEntry> for DirEntry {
 /// * The file doesn't exist.
 /// * The user lacks permissions to remove the file.
 ///
+/// This function will only ever return an error of kind `NotFound` if the given
+/// path does not exist.  Note that the inverse is not true,
```
Comment (Noratrieb) [inline library/std/src/fs.rs:1993]:
```suggestion
/// This function will only ever return an error of kind `NotFound` if the given
/// path does not exist. Note that the inverse is not true,
```

File: library/std/src/fs.rs [hunk 2]
```diff
@@ -2446,6 +2451,11 @@ pub fn create_dir_all<P: AsRef<Path>>(path: P) -> io::Result<()> {
 /// * The user lacks permissions to remove the directory at the provided `path`.
 /// * The directory isn't empty.
 ///
+/// This function will only ever return an error of kind `NotFound` if the given
+/// path does not exist.  Note that the inverse is not true,
```
Comment (Noratrieb) [inline library/std/src/fs.rs:2455]:
```suggestion
/// path does not exist. Note that the inverse is not true,
```

**Review by Noratrieb — APPROVED** [review]
did quick check of the docs for some of our platforms (Linux, FreeBSD, MacOS, Windows) and they do guarantee this

**Pull request merged**